src/Google.Api.Gax/Api/Gax/Expiration.cs

```diff
@@ -0,0 +1,76 @@
+﻿/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file or at
+ * https://developers.google.com/open-source/licenses/bsd
+ */
+
+using System;
+
+namespace Google.Api.Gax
+{
+    /// <summary>
+    /// Expiration specified by relative timeout or absolute deadline.
+    /// </summary>
+    public class Expiration
+    {
+        /// <summary>
+        /// Create an <see cref="Expiration"/> with a relative timeout.
+        /// </summary>
+        /// <param name="timeout">The relative timeout.</param>
+        /// <returns>An <see cref="Expiration"/> with the specified relative timeout.</returns>
+        /// <remarks>
+        /// Zero or negative timeouts are valid, and will cause immediate failure of the operation being performed.
+        /// </remarks>
+        public static Expiration WithTimeout(TimeSpan timeout) => new Expiration(timeout, null);
```

Comment:
"With" sounds like it's going to be an instance method to me. I'd use ForTimeout or FromTimeout. Or maybe even just Timeout.